tempodb/encoding/vparquet3/schema.go

```diff
@@ -115,11 +130,12 @@ type Attribute struct {
 	Key string `parquet:",snappy,dict"`
 
 	// This is a bad design that leads to millions of null values. How can we fix this?
-	Value        *string  `parquet:",dict,snappy,optional"`
-	ValueInt     *int64   `parquet:",snappy,optional"`
-	ValueDouble  *float64 `parquet:",snappy,optional"`
-	ValueBool    *bool    `parquet:",snappy,optional"`
-	ValueDropped string   `parquet:",snappy,optional"`
+	ValueType    AttrType  `parquet:",snappy"`
```

Comment:
> remove "this is a bad design comment"?

Done

> should we do a delta or run length encoding since most entries will be string?

I just ran a couple of tests: using `delta` instead of `snappy` is a bit larger sometimes a bit slower. Using `delta` plus `snappy` is consistently about 20% smaller than `snappy` alone

I decided to use a combination of `delta` and `snappy`